sys/riscv/riscv/swtch.S

```diff
@@ -372,6 +372,28 @@ ENTRY(cpu_switch)
 	ld	t0, TD_LOCK(a1)
 	beq	t0, s2, 1b
 #endif
+
+#ifdef CPU_QEMU_RISCV
+	/*
+	 * Check if per-thread tracing is enabled, if so pause/resume
+	 * qemu instruction tracing to reflect the new thread tracing
+	 * flag.
+	 */
+	la	t0, _C_LABEL(qemu_trace_perthread)
+	lw	t0, 0(t0)
+	beqz	t0, out_qemu_tracing
+
+	lw	t0, TD_MDFLAGS(a1)
+	andi	t0, t0, MDTD_QTRACE
+	beqz	t0, disable_qemu_tracing
+enable_qemu_tracing:
```

Comment:
Please either use anonymous 1/2/1 for these, or add .L prefixes, so that debug info doesn't get messed with.